test.sh

```diff
@@ -94,6 +94,7 @@ if printf '%s\0' "${TEST_NAMES[@]}" | grep -qwz complete_values; then
     helm upgrade --install prometheus-operator prometheus-community/kube-prometheus-stack -n "$NAMESPACE" \
         --version "60.0.2" \
         --set prometheus.prometheusSpec.serviceMonitorSelectorNilUsesHelmValues=false \
+        --set prometheus.prometheusSpec.serviceMonitorSelector.matchLabels.prometheus=default \
```

Comment:
Is this actually needed, or you added it to use the `labels` section you added in the `serviceMonitor`?

Reply:
to make sure that the labels section is actually being used

Reply:
I think `labels` is useful, because you might have different scrapers in your k8s cluster (i.e. Grafana Mimir and Prometheus) and you want only one of them to scrape these values.

Reply:
Good enough, thanks!